pull request for improvements

phatpaul opened the pull request:
Please pull these improvements.  
Some backward compatibility is broken as I added an additional parameter to some functions.

phatpaul committed:
Create README.md

pabbott-lumitec committed:
Adding work by "Pierre" Oct 23, 2013. Forked from https://code.google.com/archive/p/arduino-pwm-frequency-library/issues/2.  Modified for support for ATmega16U4/32U4 (Leonardo)

pabbott-lumitec committed:
Improvements for changing frequency on the fly.  Avoid changing pre-scaler when changing frequency, do that in the initialization instead.  Also ensure that the counter isn't already beyond the top value when changing it.

cibomahto committed:
Bugfix: missing timer for MEGA

phatpaul committed:
Merge pull request #1 from cibomahto/master

Bugfix: missing timer for MEGA

photodude commented:
I would love to see the bug fixes and the BC break separated into two PRs (if it's possible) 
Hate to see good bug fixes shelved because a BC break was introduced.

per1234 committed:
Use correct field separator in keywords.txt

The Arduino IDE requires the use of a single true tab separator between the keyword name and identifier. When spaces are used rather than a true tab, the keyword is not highlighted.

Reference:
https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification#keywords

phatpaul committed:
Merge pull request #2 from per1234/keywords-separator

Use correct field separator in keywords.txt

pabbott-lumitec committed:
add library.json for library manager to index
